src/libopenrave-core/colladaparser/colladareader.cpp

```diff
@@ -1750,13 +1750,34 @@ class ColladaReader : public daeErrorHandler
                                         if( pchild->getElementName() == std::string("controllerType") ) {
                                             jci.controllerType = pchild->getCharData();
                                         }
+                                        else if( pchild->getElementName() == std::string("robotControllerAxisProductCode") ) {
+                                            int ijointaxis = boost::lexical_cast<int>(pchild->getAttribute("axis"));
```

Comment:
this is a repeated pattern, let's cleanup.

Reply:
change the code to have ijointaxis parsed everytime getElementName() result is not `controllerType`.